eclair-core/src/main/scala/fr/acinq/eclair/crypto/Mac.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.crypto
+
+import fr.acinq.bitcoin.ByteVector32
+import org.spongycastle.crypto.digests.SHA256Digest
+import org.spongycastle.crypto.macs.HMac
+import org.spongycastle.crypto.params.KeyParameter
+import scodec.bits.ByteVector
+
+/**
+  * Created by t-bast on 04/07/19.
+  */
+
+/**
+  * Create and verify message authentication codes.
+  */
+trait Mac {
```

Comment:
I'm nitpicking, but to be precise the interface should probably be called `HMac32` and the implementation `HMacSha256` ([source](https://en.wikipedia.org/wiki/HMAC)). I know, that's not particularly elegant.

Reply:
On the contrary, this interface works for any MAC algorithm that produces a 32-bytes tag, not only for hmac. It loses generality if we call it `HMac32`.
It's true though that `Hmac256` could explicitly mention that we're using SHA (because that's what the rfc does), but it's an internal implementation detail for the users of the class so a smaller name is better isn't it?

Reply:
You are right!